### PR TITLE
Hotel sample

### PR DESCRIPTION
**Pull request by kul3r4**
@sebastianbenz PTAL

### CODE REVIEW

File: src/60_Samples_%26_Templates/Hotel.html [hunk 1]
```diff
@@ -17,279 +18,149 @@
...
-        <template amp-access-template type="amp-mustache">
-            <div class="loyalty-message">Congratulations <strong>{{username}}</strong>, your PAMPered <strong>{{status}}</strong> status has earned you <strong>{{freenights}}</strong> free nights you can apply to this stay.</div>
-        </template>
-    </section>
-</div>
-<!-- ## Dates of stay -->
+<script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
 <!--
-  An amp-form is used to allow the user to change the dates of their stay. The `type="date"` is used to allow the browser to display a native date picker. Learn more about handling the result of a form submission in the [form example](/components/amp-form).
+  We use `amp-bind` to bind the value of the room list to a variable that can change dinamically.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: suggestion: ' We use `amp-bind` to filter rooms based on their availability.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 2]
```diff
@@ -17,279 +18,149 @@
...
 <!--
-  The `amp-access` element checks the user's current session. The JSON response includes the user's loyalty status and how many free nights they have earned.
-  ```json
-  {
-    "username":"test-user",
-    "status":"Gold",
-    "freenights":2
-  }
-  ```
+  We use `amp-list` to show a list of items.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: Suggestion: We use `amp-list` to dynamically show a list of rooms.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 3]
```diff
@@ -17,279 +18,149 @@
...
-  #nearby-attractions div.theatre:before {
-    content: "🎭";
-  }
-</style>
-</head>
-<body>
-<!-- ## Hero Images -->
 <!--
-  The heading resizes to the available space using `amp-fit-text`. The `amp-carousel` element is used to add hero images for the property. To learn more about configuring these elements look at the [advanced carousel demo here](/advanced/image_galleries_with_amp-carousel/).
+  We use `amp-mustache` as a template for `amp-list`.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: s/template/template rendering engine/

File: src/60_Samples_%26_Templates/Hotel.html [hunk 4]
```diff
@@ -17,279 +18,149 @@
...
-        No rooms available.
-      </template>
-    </div>
-  </form>
-</div>
-<!-- ## Available rooms -->
+<script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js"></script>
 <!--
-  Another `amp-form` around the table allows for booking buttons.
+  We use `amp-date-picker` to implement a filter based on date selection.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: suggestion: We use `amp-date-picker` to simplify entering dates for checking room availability.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 5]
```diff
@@ -17,279 +18,149 @@
 <script async src="https://cdn.ampproject.org/v0.js"></script>
 <!-- ## Setup -->
 <!--
-Import the additional AMP components. This sample uses fitted text, an image carousel, a form, mustache templates, an iframe, and a list.
+  Import the additional AMP components. We use `amp-form` to send a filter request to the server.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: We use a few AMP extensions to dynamically render a list of rooms.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
   "preview": "default",
-  "draft": true
+  "draft": true,
+  "experiments": ["amp-date-picker", "amp-lightbox-gallery"]
   }--->
   <!--
   ## Introduction
 
-  This is a sample template for a hotel listing in AMP.
+  This is a sample showing how to implement an hotel page.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: Can you add a few more details about this sample, in particular which use cases we are covering.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 7]
```diff
@@ -17,279 +18,149 @@
...
+    .rooms [role=listitem] > * {
+      margin-right: 16px
+    }
+    .rooms [role=listitem] button {
+      margin-left: auto;
+    }
+</style>
+</head>
+<body>
+  <h1 class="m1">
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: The class=m1 should not be necessary as 

```
 body > * {
 +      margin: 16px;
 +    }
```
does the same.

Comment (sebastianbenz) [reply src/60_Samples_%26_Templates/Hotel.html]: The same applies to all other elements which are direct childs of the body.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 8]
```diff
@@ -17,279 +18,149 @@
...
-        </tr>
-      </tbody>
-    </table>
-  </form>
-</div>
-<!-- ## Embedding a map -->
+<script async custom-element="amp-date-picker" src="https://cdn.ampproject.org/v0/amp-date-picker-0.1.js"></script>
 <!--
-  A standard `amp-iframe` is used to embed a Google Map. Learn more about other iframe embeds [here](/components/amp-iframe).
+  We use `amp-lightbox-gallery` to implement a gallery of images.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: Can you improve: 'to implement a gallery of images.' - it doesn't really describe what and why we're using it.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 9]
```diff
@@ -17,279 +18,149 @@
...
+                        class="pt1 ml1">
+          <input amp-date-placeholder-start placeholder="arriving" id="arriving"/>
+          <input amp-date-placeholder-end placeholder="leaving" id="leaving"/>
+      </amp-date-picker>
+      <button class="ampstart-btn caps m1">Check Availability</button>
+    </form>
+
+    <!-- ## Additional filter -->
+    <!--
+    We implemented a filter to show only pet friendly rooms by adding a checkbox and
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: s/pet friendly/pet-friendly

File: src/60_Samples_%26_Templates/Hotel.html [hunk 10]
```diff
@@ -17,279 +18,149 @@
...
+</head>
+<body>
+  <h1 class="m1">
+    <amp-fit-text width="300" height="45" layout="responsive" max-font-size="45">
+      The Ampsterdam
+    </amp-fit-text>
+  </h1>
+  <p class="m1">The fastest hotel in town</p>
+  <h2 class="m1">When do you want to travel?</h2>
+  <!-- ## Date availability search -->
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: suggestion: Searching for available Rooms

File: src/60_Samples_%26_Templates/Hotel.html [hunk 11]
```diff
@@ -17,279 +18,149 @@
...
-      <div class="{{type}}">
-        <a href="{{url}}">{{attraction}}</a>
-      </div>
+    <amp-img width="108" height="80" src="{{img}}" lightbox alt="{{name}}-{{desc}}" class="m1"></amp-img>
+    <div class="m1">{{name}} {{#petFriendly}}&#128062;{{/petFriendly}}</div>
+    <button class="ampstart-btn caps m1 mb3">BOOK</button>
     </template>
   </amp-list>
-</div>
+    <!-- ## Rooms list Empty result -->
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: please read about title case and how to correctly use it.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 12]
```diff
@@ -17,279 +18,149 @@
...
+  <h1 class="m1">
+    <amp-fit-text width="300" height="45" layout="responsive" max-font-size="45">
+      The Ampsterdam
+    </amp-fit-text>
+  </h1>
+  <p class="m1">The fastest hotel in town</p>
+  <h2 class="m1">When do you want to travel?</h2>
+  <!-- ## Date availability search -->
+  <!--
+  You can implement a date availability search by using the `amp-date-picker` as input of a form.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: This sentence doesn't really fit into the flow. The missing part is what we want to accomplish.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 13]
```diff
@@ -17,279 +18,149 @@
...
+  Here we are using `type=range` as we need 2 dates: the arriving and leaving date of an hotel reservation.
+  Upon receiving a `submit-succes` event from the form response, we set the value of a global variable `rooms`
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: Explain why we need the `rooms` variable.

File: src/60_Samples_%26_Templates/Hotel.html (hunk 9 above)
Comment (sebastianbenz) [reply src/60_Samples_%26_Templates/Hotel.html]: Try to find a consistent way to describe the sample. In particular, first describe the why and then the how.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 14]
```diff
@@ -17,279 +18,149 @@
...
+    -->
+    <label class="ml1">Show only pet friendly rooms: 
+      <input type="checkbox" class="m1" on="change:AMP.setState({
+        showPetFriendlyOnly: event.checked
+      })">
+    </label>
+  <!-- ## Filter function -->
+  <!--
+    We use `amp-bind-macro` to implement a function which filter the `amp-list` content
+    based on the pet friendly selection filter. By using `amp-bind-macro` we avoid repeating
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: s/pet friendly/pet-friendly

Comment (sebastianbenz) [reply src/60_Samples_%26_Templates/Hotel.html]: Can you make this sentence more concise: implement a function which filter the `amp-list` content based on the pet friendly selection filter./

File: src/60_Samples_%26_Templates/Hotel.html [hunk 15]
```diff
@@ -17,279 +18,149 @@
...
+    the filter expression multiple times across the page.
+  -->
+  <amp-bind-macro id="filteredRooms"
+    expression="rooms.filter(room => showPetFriendlyOnly ? room.petFriendly : true)">
+  </amp-bind-macro>
+
+  <!-- ## Initial State -->
+  <!--
+    We are using `amp-state` to set the initial value of the variable `rooms` to the list of rooms
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: Why is this needed?

File: src/60_Samples_%26_Templates/Hotel.html [hunk 16]
```diff
@@ -17,279 +18,149 @@
...
+  <!--
+    We use `amp-bind-macro` to implement a function which filter the `amp-list` content
+    based on the pet friendly selection filter. By using `amp-bind-macro` we avoid repeating
+    the filter expression multiple times across the page.
+  -->
+  <amp-bind-macro id="filteredRooms"
+    expression="rooms.filter(room => showPetFriendlyOnly ? room.petFriendly : true)">
+  </amp-bind-macro>
+
+  <!-- ## Initial State -->
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: Can you find a better heading?

File: src/60_Samples_%26_Templates/Hotel.html [hunk 17]
```diff
@@ -17,279 +18,149 @@
...
+  <!--
+    We are using `amp-state` to set the initial value of the variable `rooms` to the list of rooms
+    returned by the server. 
+  -->
+  <amp-state id="rooms" src="https://amp-by-example-live.glitch.me/rooms"></amp-state>
+
+  <!-- ## Rooms list -->
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: use title case

File: src/60_Samples_%26_Templates/Hotel.html [hunk 18]
```diff
@@ -17,279 +18,149 @@
...
+  <!--
+    We are using `amp-list` to show a list of rooms. We bind the value of the `src` variable
+    to the result of the function `filterRooms` so that we can show the resulting list after the user selection.
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: /resulting list after the user selection/available rooms based on the user selection.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 19]
```diff
@@ -17,279 +18,149 @@
...
+
+    By clicking on an image, you can see an enlarged version: we are in fact using the `amp-lightbox-gallery`
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: s/in fact//

Comment (sebastianbenz) [reply src/60_Samples_%26_Templates/Hotel.html]: Also rephrase to improve wording.

File: src/60_Samples_%26_Templates/Hotel.html [hunk 20]
```diff
@@ -17,279 +18,149 @@
...
+    by adding the `lightbox` parameter to every `amp-img` component; the `alt` parameter content adds also
+    a caption for every image in the lightbox gallery.
+
+    `amp-list` takes all the height specified by the `height` attribute and after filtering, it could take unused space:
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Hotel.html]: improve wording

**Comment by sebastianbenz** [comment]
Sample & structure is good, documentation needs to be improved.

File: src/60_Samples_%26_Templates/Hotel.html (hunk 9 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: why  = We implemented a filter to show only pet friendly rooms
how= by adding a checkbox and setting a new variable, `filterPetFriendly`, to the value of the filter

What is missing?

File: src/60_Samples_%26_Templates/Hotel.html (hunk 7 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: it's not happening, so I am using the m1 classes

File: src/60_Samples_%26_Templates/Hotel.html (hunk 1 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 2 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 4 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 5 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 6 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 8 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 9 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 10 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 11 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 12 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 13 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 14 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 15 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 17 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 16 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 18 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 19 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

File: src/60_Samples_%26_Templates/Hotel.html (hunk 20 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done

**Comment by kul3r4** [comment]
@sebastianbenz Thanks for the feedback, please take a look at changes

File: src/60_Samples_%26_Templates/Hotel.html (hunk 3 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Hotel.html]: done
